omni-relayer/src/workers/near.rs

```diff
@@ -0,0 +1,255 @@
+                            Ok(tx_hash) => {
+                                info!("Finalized deposit: {}", tx_hash);
+                                utils::redis::remove_event(
+                                    &mut redis_connection,
+                                    &config.redis.near_sign_transfer_events,
+                                    &nonce,
+                                )
+                                .await;
+                            }
+                            Err(err) => {
+                                error!("Failed to finalize deposit: {}", err);
+                            }
+                        }
+                    }
+                }));
+            }
+        }
+
+        join_all(handlers).await;
+
+        tokio::time::sleep(tokio::time::Duration::from_secs(
+            config.redis.sleep_time_after_events_process_secs,
+        ))
+        .await;
+    }
+}
+
+pub async fn claim_fee(
+    config: config::Config,
+    redis_client: redis::Client,
+    connector: Arc<Nep141Connector>,
+) -> Result<()> {
+    let redis_connection = redis_client.get_multiplexed_tokio_connection().await?;
+
+    loop {
+        let mut redis_connection_clone = redis_connection.clone();
+        let Some(mut events) = utils::redis::get_events(
+            &mut redis_connection_clone,
+            config.redis.eth_deposit_events.clone(),
+        )
+        .await
+        else {
+            tokio::time::sleep(tokio::time::Duration::from_secs(
+                config.redis.sleep_time_after_events_process_secs,
+            ))
+            .await;
+            continue;
+        };
+
+        let mut handlers = Vec::new();
+
+        while let Some((key, event)) = events.next_item().await {
+            if let Ok(deposit_log) =
+                serde_json::from_str::<Log<crate::startup::eth::Deposit>>(&event)
+            {
+                handlers.push(tokio::spawn({
+                    let config = config.clone();
+                    let mut redis_connection = redis_connection.clone();
+                    let connector = connector.clone();
+
+                    async move {
+                        info!("Decoded log: {:?}", deposit_log);
+
+                        let Some(tx_hash) = deposit_log.transaction_hash else {
+                            log::warn!("No transaction hash in log: {:?}", deposit_log);
+                            return;
+                        };
+                        let Some(log_index) = deposit_log.log_index else {
+                            log::warn!("No log index in log: {:?}", deposit_log);
+                            return;
+                        };
+
+                        match eth_proof::get_proof_for_event(
+                            H256::from_slice(tx_hash.as_slice()),
+                            log_index,
+                            &config.mainnet.eth_rpc_http_url,
```

Comment:
Sometimes it is hardcoded mainnet, sometimes it is hardcoded testnet. Sounds strange.

Reply:
yep, this issue will go away as soon as I change config structure as you advised before